File: cindex_code/index_if.c
#include "index_if.h"
#include "term_op.h"

/*
 * openlist
 */
struct tobj* openlist(int *info)
{
    struct tobj *t;
    //int info[6];
    //printf("openlist\n");
    t = term_init(info);

    return t;
}

/*
 * close list
 */
void closelist(struct tobj* tobj)
{
    //printf("closelist \n");
    term_close(tobj);
    free(tobj);
    tobj = NULL;
}

/*
 * nextGEQ : get next docid >= input docid
 */
int nextGEQ(struct tobj* tobj, int docid)
{
    //printf("%d, next GEQ\n", docid);
    return get_docid(docid, tobj);
}

/*
 * getFreq: return current frequency
 */
int getFreq(struct tobj* tobj)
{
    //printf("get Freq \n");
    return get_freq(tobj);
}

File: query_code/index_if.py
import config
import term_op
import index_op
import cindex

""" open term 
    input term
    output termobj 
"""
def openlist(term):
    if config.CINDEX == 1:
       info = index_op.get_term_info(term)
       return cindex.openlist(info)
    else:
        return term_op.termobj(term)

""" close termobj
"""
def closelist(termobj):
    termobj.close()

""" input termobj, docid
    return next docid >= docid(input)
    if return MAXID means none
"""
def nextGEQ(termobj, docid):
    return termobj.get_docid(docid)

""" return current doc freq
    defualt first
"""
def getFreq(termobj):
    return termobj.get_freq()
